### DISCUSSION TITLE
Why does teleport only run on port 443 of IPv6 and not on IPv4?

### DISCUSSION THREAD

**Discussion by longansv** [gravitational/teleport#39049] [Q&A] [answered]
After starting teleport, I discovered that teleport only runs on port 443 of ipv6 but does not work with ipv4. So I couldn't access its web interface. Please help me
![image](https://github.com/gravitational/teleport/assets/122434297/6a335011-6adf-46ac-8aeb-3010cd205567)
![image](https://github.com/gravitational/teleport/assets/122434297/c7d09180-e421-4c20-a733-b7303e0921d0)
![image](https://github.com/gravitational/teleport/assets/122434297/bd2ba574-2bd4-493a-8d22-145ccaa5278d)

**Comment by stevenGravy** [comment]
Have you tried setting the specific IP you want instead of `0.0.0.0`? It may not be detecting it.

**Reply by longansv** [reply]
I tried but I still can't access web ui teleport via https with my domain (longhd98.click). My domain name resolution results seem to be fine 
![image](https://github.com/gravitational/teleport/assets/122434297/98066b2d-41f4-4e6b-aa12-f588456084ea)
![image](https://github.com/gravitational/teleport/assets/122434297/6b3757ef-c3ac-4c66-b5a8-2a60881fe435)

**Reply by stevenGravy** [reply]
I would first try to connect on the same machine.  `curl -k https://10.144.39.108` on the IP address you it should be listening. If that is available locally then likely the the dns connection and security groups may not be allowing it.

**Reply by longansv** [reply]
This is the result after I tested it. I don't know the cause of this error 
![image](https://github.com/gravitational/teleport/assets/122434297/c1f101a4-a66c-4785-835f-2793c7f875bf)
root@jumpserver02:~# systemctl status teleport
● teleport.service - Teleport Service
     Loaded: loaded (/lib/systemd/system/teleport.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2024-03-11 08:22:26 +07; 26s ago
   Main PID: 1775899 (teleport)
      Tasks: 14 (limit: 4515)
     Memory: 225.0M
        CPU: 5.826s
     CGroup: /system.slice/teleport.service
             └─1775899 /usr/local/bin/teleport start --config /etc/teleport.yaml --pid-file=/run/teleport.pid

Mar 11 08:22:47 jumpserver02 teleport[1775899]:         runtime/asm_amd64.s:1650 runtime.goexit
Mar 11 08:22:47 jumpserver02 teleport[1775899]: User Message: acme/autocert: missing certificate] alpnproxy/proxy.go:339
Mar 11 08:22:47 jumpserver02 teleport[1775899]: 2024-03-11T08:22:47+07:00 WARN [ALPN:PROX] Failed to handle client connection. error:[
Mar 11 08:22:47 jumpserver02 teleport[1775899]: ERROR REPORT:
Mar 11 08:22:47 jumpserver02 teleport[1775899]: Original Error: *errors.errorString acme/autocert: missing certificate
Mar 11 08:22:47 jumpserver02 teleport[1775899]: Stack Trace:
Mar 11 08:22:47 jumpserver02 teleport[1775899]:         github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:394 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).handleConn
Mar 11 08:22:47 jumpserver02 teleport[1775899]:         github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:328 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).Serve.func1
Mar 11 08:22:47 jumpserver02 teleport[1775899]:         runtime/asm_amd64.s:1650 runtime.goexit
Mar 11 08:22:47 jumpserver02 teleport[1775899]: User Message: acme/autocert: missing certificate] alpnproxy/proxy.go:339
``

```

**Reply by longansv** [reply]
Thank you for your help. I followed this guide and it was successfully fixed. Error related to acme ssl https://github.com/gravitational/teleport/issues/29863

**Comment by webvictim** [comment, marked as answer]
`tcp6` is dual-stacked on modern kernels, meaning it listens on both IPv4 and IPv6 at once.

Bear in mind Teleport uses HTTPS, so you should be connecting to https://longhd98.click

**Reply by longansv** [reply]
yes I visited https://longhd98.click but the result is still inaccessible

**Reply by longansv** [reply]
Thank you for your help. I followed this guide and it was successfully fixed. Error related to acme ssl https://github.com/gravitational/teleport/issues/29863
